PocketKit/Sources/PocketKit/ItemAction.swift

```diff
@@ -0,0 +1,125 @@
+import Foundation
+import Textile
+import UIKit
+
+
+struct ItemAction {
+    let title: String
+    let identifier: UIAction.Identifier
+    let accessibilityIdentifier: String
+    let image: UIImage?
+    let handler: (() -> ())?
+}
+
+extension ItemAction: Hashable {
+    static func == (lhs: ItemAction, rhs: ItemAction) -> Bool {
+        lhs.title == rhs.title
+        && lhs.accessibilityIdentifier == rhs.accessibilityIdentifier
+        && lhs.image == rhs.image
+    }
+
+    func hash(into hasher: inout Hasher) {
+        hasher.combine(title)
+        hasher.combine(accessibilityIdentifier)
+        hasher.combine(image)
+    }
+}
+
+extension ItemAction {
+    static func save(_ handler: @escaping () -> ()) -> ItemAction {
+        return ItemAction(
+            title: "Save",
+            identifier: .saveItem,
+            accessibilityIdentifier: "item-action-save",
+            image: UIImage(asset: .save),
+            handler: handler
+        )
+    }
+
+    static func archive(_ handler: @escaping () -> ()) -> ItemAction {
+        return ItemAction(
+            title: "Archive",
+            identifier: .archiveItem,
+            accessibilityIdentifier: "item-action-archive",
+            image: UIImage(asset: .archive),
+            handler: handler
+        )
+    }
+
+    static func delete(_ handler: @escaping () -> ()) -> ItemAction {
+        return ItemAction(
+            title: "Delete",
+            identifier: .deleteItem,
+            accessibilityIdentifier: "item-action-delete",
+            // TODO: Use the pocket trash can icon here
+            image: UIImage(systemName: "trash"),
+            handler: handler
+        )
+    }
+
+    static func favorite(_ handler: @escaping () -> ()) -> ItemAction {
+        return ItemAction(
+            title: "Favorite",
+            identifier: .favoriteItem,
+            accessibilityIdentifier: "item-action-favorite",
+            image: UIImage(asset: .favorite)
+                .withTintColor(UIColor(.ui.grey5), renderingMode: .alwaysOriginal),
+            handler: handler
+        )
+    }
+
+    static func unfavorite(_ handler: @escaping () -> ()) -> ItemAction {
+        return ItemAction(
+            title: "Unfavorite",
+            // intentionally the same as `favorite()` since we want to replace
+            identifier: .favoriteItem,
+            accessibilityIdentifier: "item-action-favorite",
+            image: UIImage(asset: .favoriteFilled)
+                .withTintColor(UIColor(.branding.amber4), renderingMode: .alwaysOriginal),
+            handler: handler
+        )
+    }
+
+    static func share(_ handler: @escaping () -> ()) -> ItemAction {
+        return ItemAction(
+            title: "Share",
+            // intentionally the same as `favorite()` since we want to replace
```

Comment:
Perhaps this is a byproduct of a copy/paste for (un)favorite actions.

Reply:
yup